tokio/src/process/unix/mod.rs

```diff
@@ -203,22 +194,22 @@ where
     }
 }
 
-pub(crate) type ChildStdin = PollEvented<Fd<std::process::ChildStdin>>;
-pub(crate) type ChildStdout = PollEvented<Fd<std::process::ChildStdout>>;
-pub(crate) type ChildStderr = PollEvented<Fd<std::process::ChildStderr>>;
+pub(crate) type ChildStdin = PollEvented<Pipe>;
+pub(crate) type ChildStdout = PollEvented<Pipe>;
+pub(crate) type ChildStderr = PollEvented<Pipe>;
 
-fn stdio<T>(option: Option<T>) -> io::Result<Option<PollEvented<Fd<T>>>>
+fn stdio<T>(option: Option<T>) -> io::Result<Option<PollEvented<Pipe>>>
 where
-    T: AsRawFd,
+    T: IntoRawFd,
 {
     let io = match option {
         Some(io) => io,
         None => return Ok(None),
     };
 
     // Set the fd to nonblocking before we pass it to the event loop
-    unsafe {
-        let fd = io.as_raw_fd();
+    let fd = unsafe {
+        let fd = io.into_raw_fd();
```

Comment:
This will leak the fd if we bail out due to a libc error below (since we don't take ownership of the fd until those checks are done). I'll submit a fix